administrator/components/com_menus/views/items/view.html.php

```diff
@@ -219,8 +219,25 @@ public function display($tpl = null)
 
 		$this->f_levels = $options;
 
-		$this->addToolbar();
-		$this->sidebar = JHtmlSidebar::render();
+		// We don't need toolbar in the modal window.
+		if ($this->getLayout() !== 'modal')
+		{
+			$this->addToolbar();
+			$this->sidebar = JHtmlSidebar::render();
+		}
+		else
+		{
+			// In menu items associations modal we need to remove language filter if forcing a language.
+			if ($forcedLanguage = JFactory::getApplication()->input->get('forcedLanguage', '', 'CMD'))
```

Comment:
Then it's a good time to improve that and make it better ;-) *duck

Keep up your good work!